pkg/security/jwt/validator.go

```diff
@@ -34,6 +90,82 @@ func errParseTokenInvalidClaimsType(t *jwt.Token) error {
 	return fmt.Errorf("%w: unsupported type %T", ErrCannotParseToken, t.Claims)
 }
 
+func (v *Validator) checkTrust(ctx context.Context, token string, claims jwt.Claims) error {
+	issuer, err := claims.GetIssuer()
+	if err != nil {
+		return err
+	}
+	issuer = pkgHttpUri.CanonicalURI(issuer)
+
+	client, ok := v.clients[issuer]
+	if !ok {
+		v.logger.Debugf("client not set for issuer %v, trust verification skipped", issuer)
+		return nil
+	}
+
+	tokenID, err := getID(claims)
+	if err != nil {
+		return err
+	}
+
+	tr, ok := v.tokenCache.Load(tokenID, issuer)
+	if ok {
+		if tr.Blacklisted {
+			return ErrBlackListedToken
+		}
+		return nil
+	}
+
+	uri, err := url.Parse(client.tokenEndpoint)
```

Comment:
until the token is not in cache you will make each request even the token is same. Make only one request for one token in same time.